driver/others/memory.c

```diff
@@ -415,23 +417,72 @@ int openblas_get_num_threads(void) {
 #endif
 }
 
-struct release_t {
-  void *address;
-  void (*func)(struct release_t *);
-  long attr;
-};
-
 int hugetlb_allocated = 0;
 
 #if defined(OS_WINDOWS)
 #define THREAD_LOCAL __declspec(thread)
-#define UNLIKELY_TO_BE_ZERO(x) (x)
+#define LIKELY_ONE(x) (x)
 #else
 #define THREAD_LOCAL __thread
-#define UNLIKELY_TO_BE_ZERO(x) (__builtin_expect(x, 0))
+#define LIKELY_ONE(x) (__builtin_expect(x, 1))
+#endif
+
+struct alloc_t {
+  int used;
+  int attr;
+  void (*func)(struct alloc_t *);
+  /* Pad to 64-byte alignment */
+#ifdef __64BIT__
+  char pad[48];
```

Comment:
This seems fragile, is there no way for the compiler to do the padding automatically, or by using a macro of some sort here which will deduce the size and do the right thing?

Reply:
Done.  Also added more documentation to the struct.